app/screens/settings-screen/account/settings/email.tsx

```diff
@@ -0,0 +1,191 @@
+  const [setEmailMutation, { loading: emRegLoading }] =
+    useUserEmailRegistrationInitiateMutation()
+
+  const deleteEmail = async () => {
+    try {
+      await emailDeleteMutation()
+      toastShow({
+        type: "success",
+        message: LL.AccountScreen.emailDeletedSuccessfully(),
+        LL,
+      })
+    } catch (err) {
+      Alert.alert(LL.common.error(), err instanceof Error ? err.message : "")
+    }
+  }
+
+  const deleteEmailPrompt = async () => {
+    Alert.alert(
+      LL.AccountScreen.deleteEmailPromptTitle(),
+      LL.AccountScreen.deleteEmailPromptContent(),
+      [
+        { text: LL.common.cancel(), onPress: () => {} },
+        {
+          text: LL.common.yes(),
+          onPress: async () => {
+            deleteEmail()
+          },
+        },
+      ],
+    )
+  }
+
+  const tryConfirmEmailAgain = async (email: string) => {
+    try {
+      await emailDeleteMutation({
+        // to avoid flacky behavior
+        // this could lead to inconsistent state if delete works but set fails
+        fetchPolicy: "no-cache",
+      })
+
+      const { data } = await setEmailMutation({
+        variables: { input: { email } },
+      })
+
+      const errors = data?.userEmailRegistrationInitiate.errors
+      if (errors && errors.length > 0) {
+        Alert.alert(errors[0].message)
+      }
+
+      const emailRegistrationId = data?.userEmailRegistrationInitiate.emailRegistrationId
+
+      if (emailRegistrationId) {
+        navigate("emailRegistrationValidate", {
+          emailRegistrationId,
+          email,
+        })
+      } else {
+        console.warn("no flow returned")
+      }
+    } catch (err) {
+      console.error(err, "error in setEmailMutation")
+    }
+  }
+
+  const reVerifyEmailPrompt = () => {
+    if (!email) return
+    Alert.alert(
+      LL.AccountScreen.emailUnverified(),
+      LL.AccountScreen.emailUnverifiedContent(),
+      [
+        { text: LL.common.cancel(), onPress: () => {} },
+        {
+          text: LL.common.ok(),
+          onPress: () => tryConfirmEmailAgain(email),
+        },
+      ],
+    )
+  }
+
+  return (
+    <SettingsRow
+      loading={loading}
+      spinner={emDelLoading || emRegLoading}
+      title={title(email, emailVerified, LL)}
+      subtitle={emailVerified ? email?.toString() : email}
+      leftIcon="mail-outline"
+      action={email ? null : () => navigate("emailRegistrationInitiate")}
+      rightIcon={
+        email ? (
+          <View style={styles.sidetoside}>
+            {!emailVerified && (
+              <GaloyIconButton
+                name="refresh"
+                size="medium"
+                onPress={reVerifyEmailPrompt}
+              />
+            )}
+            {(bothEmailAndPhoneVerified || (email && !emailVerified)) && (
+              <GaloyIconButton name="close" size="medium" onPress={deleteEmailPrompt} />
+            )}
+          </View>
+        ) : undefined
```

Comment:
maybe this logic should not be in the JSX part

Reply:
true, looks convoluted in my eyes as well